rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -91,12 +119,17 @@ impl ArrowBlockfileWriter {
             sparse_index: new_sparse_index,
             id,
             write_mutex: Arc::new(tokio::sync::Mutex::new(())),
+            options,
         }
     }
 
-    pub(crate) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
+    pub(crate) async fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
```

Comment:
I discovered `commit()` does not actually need to be async after going through and updating all usage 😅
open to reverting this, but is there a future scenario where it would have to be async?

Reply:
Yes, if you wanted to stage any I/O to allow commit() to write to disk without flushing etc